### PR TITLE
ua-nodeset: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**
Specify library name and version:  **ua-nodeset/***

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ua-nodeset/all/conanfile.py [hunk 1]
```diff
@@ -10,34 +12,36 @@ class UaNodeSetConan(ConanFile):
...
-        license_contents = content[2:content.find("*/", 1)]
-        tools.save("LICENSE", license_contents)
+    def layout(self):
+        pass
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+    def package_id(self):
+        del self.info.settings.compiler
+        del self.info.settings.build_type
```
Comment (uilianries) [inline recipes/ua-nodeset/all/conanfile.py:25]: why does it require arch and os, if there is no build and the conandata.yml has no distinction about it?

File: recipes/ua-nodeset/all/conanfile.py [hunk 2]
```diff
@@ -10,34 +12,36 @@ class UaNodeSetConan(ConanFile):
...
     url = "https://github.com/conan-io/conan-center-index"
-    topics = ("opc-ua-specifications", "uanodeset", "normative-files", "companion-specification")
-
-    no_copy_source = True
+    topics = ("opc-ua-specifications", "uanodeset", "normative-files", "companion-specification", "pre-built")
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    package_type = "application"
```
Comment (uilianries) [inline recipes/ua-nodeset/all/conanfile.py:17]:
```suggestion
    package_type = "build-scripts"
```
The package does not contain a single executable file, but only consumable files .xml, .csv, .bsd, .ua ...

File: recipes/ua-nodeset/all/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/ua-nodeset/all/conanfile.py]: Copied this from the pre-built package template initially and forgot to update it. Thanks for catching this!

File: recipes/ua-nodeset/all/conanfile.py (hunk 2 above)
Comment (valgur) [reply recipes/ua-nodeset/all/conanfile.py]: Thanks! I hope this fixes the linter issue.

**Review by uilianries — COMMENTED** [review]
@valgur It's looking good, I just recommend you simplifying the test package instead, please, take a look on https://github.com/valgur/conan-center-index/pull/1

**Review by uilianries — APPROVED** [review]
LGTM

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Pull request merged**